pkg/lib/harness/llm-harness.go

```diff
@@ -181,10 +183,35 @@ func PrintLogs(configHome string, w io.Writer) error {
 		return fmt.Errorf("error reading log file: %w", err)
 	}
 	defer logFile.Close()
-	if _, err = io.Copy(w, logFile); err != nil {
-		return fmt.Errorf("failed to print log file: %w", err)
+	reader := bufio.NewReader(logFile)
+	for {
+		line, err := reader.ReadString('\n')
+		if err != nil {
+			if err == io.EOF {
+				if !follow {
+					return nil
+				}
+				time.Sleep(1 * time.Second)
+				serverStatus := checkExistence(configHome)
+				if serverStatus {
+					return fmt.Errorf("server stopped")
+				}
+				continue
+			} else {
+				return fmt.Errorf("failed to print log file: %w", err)
+			}
+		}
+		if _, err := w.Write([]byte(line)); err != nil {
+			return fmt.Errorf("failed to write to output: %w", err)
+		}
 	}
-	return nil
+}
+
+func checkExistence(configHome string) bool {
+	pidFile := filepath.Join(constants.HarnessPath(configHome), constants.HarnessProcessFile)
+
+	_, err := readPIDFromFile(pidFile)
+	return os.IsNotExist(err)
 }
```

Comment:
This is just checking if the PID file exists, not whether the server is actively running. I think we also need to use `isProcessRunning(pid)`

Reply:
When we run `kit dev stop` we are verifying the the running server this way, and stopping it. Used the same logic in here.

Reply:
`func (harness *LLMHarness) Stop() error {
	pidFile := filepath.Join(constants.HarnessPath(harness.ConfigHome), constants.HarnessProcessFile)

	pid, err := readPIDFromFile(pidFile)
	if os.IsNotExist(err) {
		return fmt.Errorf("no Running server found")
	}
	if err != nil {
		return err
	}
`

This is how we are checking the server status in stop interface.

Reply:
We still need to show logs even if server crashes. I think including the above case wouldn't display logs when the server crashes. We only remove pid file once server ends properly until then we display logs.

Reply:
These checks run only when reaching EOF on the file though -- we've already printed all existing logs at this point. If the server has crashed, there aren't going to be any more logs to print (same as the server terminated normally case)

Reply:
Okay, will add that case too.